chain-signatures/node/src/mesh/connection.rs

```diff
@@ -207,11 +210,18 @@ impl Pool {
         let mut stable = Vec::new();
         let mut active = Participants::default();
         for (participant, conn) in self.connections.iter() {
-            if let NodeStatus::Active(is_stable) = conn.status().await {
-                active.insert(participant, conn.info.clone());
-                if is_stable {
+            match conn.status().await {
+                NodeStatus::Active => {
+                    active.insert(participant, conn.info.clone());
                     stable.push(*participant);
                 }
+                NodeStatus::Inactive => {
+                    // TODO: should we really insert inactive nodes to the active list here?
+                    // For now, in the refactoring PR, I just keep the exact same behavior.
+                    // We can delete this line when in the next PR.
+                    active.insert(participant, conn.info.clone());
+                }
```

Comment:
To keep this a pure refactor, I kept the old behavior here. These are all nodes that respond in the `/state` endpoint that they are not in `NodeState::Running` nor in `NodeState::Resharing`. Meaning, they are in one of  { `Starting`, `Started`, `Generating`, `WaitingForConsensus`, `Joining` }.


@ChaoticTempest can you advise what the best treatment would be? For me it would be more intuitive to keep nodes that are currently joining the network outside the "active" participants set. Only once the are in running or resharing state I would add them. But I'm not sure if it will break something.

Reply:
Yeah that would make sense to keep them separate. I think we need to do a refactor on what these sets are. When the node is initially doing keygen at the genesis of the network, they use `{ Starting, Started, Generating, WaitingForConsensus, Joining }` as active set. We can definitely be more fine grained with this and take this out of the active set just in case we're interfering with protocol generation side.

But outside of that, I think it's good to add in this now and make keygen use inactive set later